### ISSUE TITLE
Axi_CDMA (Central Direct Memory Access) device driver

### ISSUE THREAD

**Issue by 92jisuk** [bperez77/xilinx_axidma#73]
Hi. I am happy to see you in your github.

I found the device driver of axi_dma in your github.
I ported DMA driver on the ZED board using your code.
It was working very well.

But now, I am trying to port the CDMA driver on my board.
I must add and change some codes for porting the CDMA driver.
But I have no idea for it...

Could you share the CDMA device driver codes if you have....?

Thank you.

**Comment by 92jisuk** [comment]
Hi. I searched about CDMA driver in Google.
And I found the cdmatest.c code what looks like device driver of CDMA.
Here is the github site: https://github.com/Xilinx/linux-xlnx/blob/master/drivers/dma/xilinx/cdmatest.c

Could you check this is the correct code for axi_cdma device driver?
If right, What should I do using this code...?

Thank you.

**Comment by bperez77** [comment]
Certainly, I don't have the code right now, but I should be able to implement it in a few days. I'm currently away from my FPGA board, so the code won't be heavily tested, but I think it should be straight-forward.

The CDMA test driver contains the basic template for setting up a CDMA transaction in kernel code, but it doesn't provide the same functionality as this driver. It simply sends some arbitrary data and expects a loopback design (connecting the TX channel to the RX channel to simply loop the data back to the CPU), but it can be used to test your HW design.

**Comment by 92jisuk** [comment]
Hi Brandon.
 
Thank you for consideration about my issue.
I can implement using AxiDMA instead of AxiCDMA eventhough it is inconvenient for me.
 
If making the cdma device driver is not for your job, I don't want to steal your time...
I hope to really  appreciate for your mind.
 
Thank you Brandon!
 
From Jisu.
 
-----Original Message-----

**Comment by 92jisuk** [comment]
Hi, Brandon.
I said that you don't need to effort for my question because I had thought that you are very busy...
But I need your help...

I got a "testcdma.c" code from:
https://github.com/Xilinx/linux-xlnx/blob/master/drivers/dma/xilinx/cdmatest.c
and tried to compile this file using petalinux to get a CDMA device driver .
I failed to compile it with some errors and I am wondering this file is exact device driver file or not.

Could you please help me to create CDMA device driver?
Thank you.

From Jisu
